routes/jobRoutes.js

```diff
@@ -1,155 +1,220 @@
-    // logger.info("job/<>/history");
-
-    try{
-        var query = {jobId: req.params.jobId};
-        if(req.query.version) query["validity.version"] = parseInt(req.query.version);
-
-        var options = {};
-        let [job,processes,versions] = await Promise.all([
-            Jobs.retrieve(query),
-            Processes.findInputRecord(req.params.job_id),
-            Jobs.versions(req.params.jobId),
-          ]);
-        if(!job) return res.status(404).send("No such job recorded.");
-        var formId = job.formId;
-        if(!formId) throw new Error("Job has no formId");
-
-        if(req.query.formVersion) options.version = parseInt(req.query.formVersion);
-        // fixme rollback
-        var formrec = await Forms.retrieve('jobForms',formId,options);
-        var formVersions = await Forms.getFormVersions('jobForms',formId);
-        // logger.info('versions',versions);
-        if(!formrec) return res.status(400).send("No such job form");  
-        res.render('viewJobHistory.pug',{formId:req.params.formId, formrec, processes, job, versions, formVersions, retrieved:true})
-    } 
-    catch(err) { 
-      logger.error("error in router function",err); next(); 
+// Submit a new job of a given type
+router.get('/job/:formId', permissions.checkPermission("jobs:submit"), async function(req, res, next)
+{
+  try
+  {
+    // Retrieve the job type form corresponding to the specified form ID
+    // Set the retrieval (and therefore, submission) timestamp to be the current date and time
+    // Throw an error if there is no DB entry corresponding to this form ID
+    var form = await Forms.retrieve('jobForms', req.params.formId, {onDate: new Date()});
+    
+    if(!form)
+    {
+      return res.status(400).send("There is no DB entry for a job type form with ID: " + formId);
     }
-
+    
+    // Render the page for submitting a new job
+    res.render('job_submit.pug', {form});
+  }
+  catch(err)
+  {
+    res.status(400).send(err.toString());
+  }
 });
 
 
-
-/// Run an new job
-router.get("/job/:formId",permissions.checkPermission("jobs:submit"),async function(req,res,next){
-  try{
-    // logger.info("run a new job");
-    var options = {onDate: new Date()};
-    var workflow = await Forms.retrieve('jobForms',req.params.formId,options);
-    if(!workflow) return res.status(400).send("No such job workflow");
-    res.render('test_run_singleComponent.pug',{formId:req.params.formId, form:workflow, 
-                          route_on_submit: '/job',
-                          submission_url: '/json/job'});
-  } catch(err) { logger.error(err); next(); }
+// Edit an existing job
+router.get('/job/:jobId([A-Fa-f0-9]{24})/edit', permissions.checkPermission("jobs:submit"), async function(req, res, next)
+{
+  try
+  {
+    // Retrieve the job's DB entry, using its job ID
+    // Throw an error if there is no DB entry corresponding to this job ID
+    var job = await Jobs.retrieve(req.params.jobId);
+    
+    if(!job)
```

Comment:
`if (!job) next();` is more succinct

Reply:
Mmm, personally I still prefer even single-line statements to be indented in brackets, same as multi-line blocks.  Feel like it keeps things more consistent.  Stylistic differences?